src/PdfSharper/Pdf.Advanced/PdfCrossReferenceStream.cs

```diff
@@ -49,11 +53,39 @@ public PdfCrossReferenceStream(PdfDocument document)
                 Debug.WriteLine("PdfCrossReferenceStream created.");
             }
 #endif
+
+            if (initialize)
+            {
+                Initialize();
+            }
+        }
+
+        private void Initialize()
+        {
+            Owner.UnderConstruction = true;
+            try
+            {
+                PdfDictionary decodeParams = new PdfDictionary(Owner);
+                decodeParams.IsCompact = true;
+                decodeParams.Elements.SetInteger("/Columns", 3);
+                decodeParams.Elements.SetInteger("/Predictor", 12);
+
+                Elements.SetObject(PdfDictionary.PdfStream.Keys.DecodeParms, decodeParams);
+                Elements.SetName(Keys.Type, "XRef");
+                PdfArray widthesArray = new PdfArray(Owner, new PdfInteger(1), new PdfInteger(1), new PdfInteger(1));
+                Elements.SetObject(Keys.W, widthesArray);
+
+                Stream = new PdfStream(this);
+            }
+            finally
+            {
+                Owner.UnderConstruction = false;
+            }
         }
 
-        public readonly List<CrossReferenceStreamEntry> Entries = new List<CrossReferenceStreamEntry>();
+        public List<CrossReferenceStreamEntry> Entries { get; private set; } = new List<CrossReferenceStreamEntry>();
```

Comment:
I learn something everyday.. this must be new new feature that we talked about the other day.